Content.Server/Research/Systems/ResearchSystem.Server.cs

```diff
@@ -96,7 +102,8 @@ public int PointsPerSecond(ResearchServerComponent component)
     {
         var points = 0;
 
-        if (CanRun(component))
+        // Is our machine powered, and are we below our limit of passive point gain?
+        if (CanRun(component) && component.Points < (component.PassiveLimitPerSource * component.PointSources.Count))
```

Comment:
With this logic it's possible for points to go over the limit slightly if there are enough point sources to take it from <30k to >30k in one second.

Reply:
doesn't really effect anything, the techs are all multiples of 1000. It's also intended that you can go over with research disks, too

Reply:
Yeah, this bug is only possible when there is more than one point source, and I'm pretty sure sci research is getting reworked anyway which will probably get rid of point sources entirely.

Reply:
it happens live when centcomm loads